src/main/groovy/org/gradle/android/AndroidCacheFixPlugin.groovy

```diff
@@ -55,14 +55,10 @@ class AndroidCacheFixPlugin implements Plugin<Project> {
 
     @Override
     void apply(Project project) {
-        workarounds.addAll(initializeWorkarounds(project))
+        workarounds.addAll(initializeWorkarounds())
 
-        if (!isSupportedAndroidVersion(project)) {
-            if (isMaybeSupportedAndroidVersion(project)) {
-                Warnings.MAYBE_SUPPORTED_ANDROID_VERSION.warnOnce(project)
-            } else {
-                throw new RuntimeException("Android plugin ${CURRENT_ANDROID_VERSION} is not supported by Android cache fix plugin. Supported Android plugin versions: ${SUPPORTED_ANDROID_VERSIONS.join(", ")}. Please check if a newer version of this plugin is available or override with -D${IGNORE_VERSION_CHECK_PROPERTY}=true.")
-            }
+        if (!isMaybeSupportedAndroidVersion()) {
```

Comment:
What does it mean to be "maybe supported"? Something should be either supported or not. Is there a scenario where this returns `true` and we later find out the version is not supported?

Reply:
This is not something introduced by these changes/pr. 
The explanation "maybe supported" comes from the idea that release cycles of the AGP won't include api changes in patch releases. Our mechanism to apply workarounds in the project build classpath is based on compatibility with major-minor AGP versions, hence the previous approach to throw a warning when the version didn't match with major-minor-patch.  
If it's not clear, we can rename the function to isSupportedAndroidVersion

Reply:
exactly, some of the workarounds are fixed, each workaround contains the annotation `AndroidIssue` where you can track when it was fixed in the AGP. Example:
https://github.com/gradle/android-cache-fix-gradle-plugin/blob/main/src/main/groovy/org/gradle/android/workarounds/BundleLibraryClassesWorkaround.groovy#L13C2-L13C14

Reply:
The maybe confused me too. I think we should drop it.

Reply:
done